### PR TITLE
[YAML] Java Filter and explode operations.

### PR DESCRIPTION
**Pull request by robertwb**
This is a follow-up on top of https://github.com/apache/beam/pull/28657

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/transforms/providers/JavaExplodeTransformProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas.transforms.providers;
+
+import static org.junit.Assert.*;
+
+import java.util.List;
```
Comment (Polber) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/transforms/providers/JavaExplodeTransformProviderTest.java:22]: I assume beam follows the no star import convention but I could be wrong

File: sdks/python/apache_beam/yaml/standard_providers.yaml
Comment (Polber) [inline sdks/python/apache_beam/yaml/standard_providers.yaml:1]: Don't `Filter` and `Explode` need to be specified here?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/providers/JavaExplodeTransformProvider.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  }
+
+  @Override
+  protected SchemaTransform from(Configuration configuration) {
+    return new ExplodeTransform(configuration);
+  }
+
+  @Override
+  public String identifier() {
+    return String.format("beam:schematransform:org.apache.beam:yaml:map_to_fields-java:v1");
```
Comment (Polber) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/providers/JavaExplodeTransformProvider.java:70]: Same URN as `MapToFields`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/providers/JavaFilterTransformProvider.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  }
+
+  @Override
+  protected SchemaTransform from(Configuration configuration) {
+    return new JavaFilterTransform(configuration);
+  }
+
+  @Override
+  public String identifier() {
+    return String.format("beam:schematransform:org.apache.beam:yaml:map_to_fields-java:v1");
```
Comment (Polber) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/providers/JavaFilterTransformProvider.java:68]: Same URN as above

**Review by Polber — CHANGES_REQUESTED** [review]

File: sdks/python/apache_beam/yaml/standard_providers.yaml
Comment (robertwb) [reply sdks/python/apache_beam/yaml/standard_providers.yaml:1]: Fixed.

**Review by Polber — COMMENTED** [review]
LGTM. Should probably wait until #28657 is merged to make sure there are no conflicts

**Review by Polber — APPROVED** [review]

**Pull request merged**